cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -283,6 +285,19 @@ bool ExecNode::ErrorIfNotOk(Status status) {
   return true;
 }
 
+Status ExecNode::SubmitTask(std::function<Status()> task) {
+  auto executor = plan()->exec_context()->executor();
+  auto maybe_future = executor->Submit(std::move(task));
+  if (!maybe_future.ok()) {
+    outputs_[0]->ErrorReceived(this, maybe_future.status());
```

Comment:
Nit: It doesn't matter today but if this is a general method we may want to loop through all outputs.